shell/src/main/java/alluxio/cli/bundler/command/CollectLogCommand.java

```diff
@@ -14,21 +14,121 @@
 import alluxio.client.file.FileSystemContext;
 import alluxio.conf.PropertyKey;
 import alluxio.exception.AlluxioException;
+import alluxio.util.CommonUtils;
 
+import jline.internal.Nullable;
 import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
 import org.apache.commons.io.FileUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.io.File;
+import java.io.FileNotFoundException;
 import java.io.IOException;
+import java.net.URI;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 /**
  * Command to collect Alluxio logs.
  * */
 public class CollectLogCommand  extends AbstractCollectInfoCommand {
   public static final String COMMAND_NAME = "collectLog";
   private static final Logger LOG = LoggerFactory.getLogger(CollectLogCommand.class);
+  public static final Set<String> FILE_NAMES = Stream.of(
+      "master.log",
+      "master.out",
+      "job_master.log",
+      "job_master.out",
+      "master_audit.log",
+      "worker.log",
+      "worker.out",
+      "job_worker.log",
+      "job_worker.out",
+      "proxy.log",
+      "proxy.out",
+      "task.log",
+      "task.out",
+      "user"
+  ).collect(Collectors.toSet());
+  // We tolerate the beginning of a log file to contain some rows that are not timestamped.
+  // 30 is chosen because a YARN application log can have >20 rows in the beginning for
+  // general information about a job.
+  // The timestamped log entries start after this general information block.
+  private static final int TRY_PARSE_LOG_ROWS = 30;
+
+  // Preserves the order of iteration, we try the longer pattern before the shorter one
+  private static final Map<String, Integer> FORMAT_TO_LEN = new LinkedHashMap<String, Integer>(){
```

Comment:
Why can't this just be a list? Why do we need to cache the length?

Reply:
We need the length because the `DateTimeFormatter` CANNOT parse anything other than the datetime. For example a DateTimeFormatter of format "yyyy-mm-dd" cannot parse "2020-07-15 INFO i did something".  I have to cut the target string before parsing. 

One example of existing wheels is `org.apache.commons.lang.time.DateUtils.parse(String s)`. It builds its own state machine but still cannot avoid the need to index and cut the string `s`.

All in all, for each format I try, i need a DateTimeFormatter and a length of how much string I need. I've moved these two into one `LinkedHashMap` now, instead of using two separated maps.